ng4mt.py

```diff
@@ -0,0 +1,144 @@
+import nevergrad as ng
+
+import sys
+import subprocess
+import json
+import argparse
+
+from mpi4py import MPI
+from mpi4py.futures import MPIPoolExecutor
+from multiprocessing import freeze_support, cpu_count
+
+
+def get_sf_parameters(stockfish_exe):
+    """Run sf to obtain the tunable parameters"""
+
+    process = subprocess.Popen(
+        stockfish_exe, shell=True, stdin=subprocess.PIPE, stdout=subprocess.PIPE
+    )
+    output = process.communicate(input=b"quit\n")[0]
+    if process.returncode != 0:
+        sys.stderr.write("get_sf_parameters: failed to execute command: %s\n" % command)
+        sys.exit(1)
+
+    # parse for parameter output
+    params = ng.p.Dict()
+    for line in output.decode("utf-8").split("\n"):
+        if "Stockfish" in line or not "," in line:
+            continue
+        fields = line.split(",")
+        params[fields[0]] = ng.p.Scalar(
+            init=int(fields[1]), lower=int(fields[2]), upper=int(fields[3])
+        ).set_integer_casting()
+
+    return params
+
+
+class Matetrack:
+    def __init__(self, engine, nodes, concurrency, epdFile):
+        self.engine = engine
+        self.nodes = nodes
+        self.concurrency = concurrency
+        self.epdFile = epdFile
+
+    def call(self, d):
+        cmd = [
+            "python",
+            "matecheck.py",
+            "--engine",
+            self.engine,
+            "--epdFile",
+            self.epdFile,
+            "--nodes",
+            str(self.nodes),
+            "--concurrency",
+            str(self.concurrency),
+            "--engineOpts",
+            json.dumps(d),
+        ]
+        output = subprocess.run(
+            cmd,
+            capture_output=True,
+            text=True,
+        )
+        if output.returncode != 0:
+            print("Executing matecheck failed....")
+            print(cmd)
+            print("yields: ")
+            print(output.stderr)
+            print("after: ")
+            print(output.stdout)
+            sys.exit(1)
+
+        for line in output.stdout.split("\n"):
+            # we minimize, so return minus the number of best mates
+            if "Best mates:  " in line:
+                return -int(line.split()[2])
```

Comment:
Could also subtract a small multiple of mates found?

Reply:
In any case I would also add `"Better mates:"`, in case the `bm` value from the epd file is not (yet) optimal. In that case could also log some message with relevant FEN.